### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by ninoCookies** [jeffcarp/braintree-angular#37]
I was wondering.. Is the dropin example production ready ? (Once one enteres all the necessary data?) As far as i understood braintree documentation the token is not needed for dropin situation so this example should actually be good enough for production, am i right ?

Also how would you receive the clientTokenPath constant if the server.js is running on different server for example ? Atm i have nodejs server running in different project as the client form. I tried http request but i cannot inject the response into constant.. Am i missing something ?

Great work btw, thank you.

**Comment by ninoCookies** [comment]
Ok so here is an update to second part of my question :D

I hosted my server.js on a remote VPS and added a http request to clientTokenPath - when i did that i received CORS - Cross Origin Resource Sharing problems. So i used https://www.npmjs.com/package/cors. Now i have server.js running on remote server and client connecting to it. 

My first question still remains tho... Is it possible to push this to production (from safety point of view ?)

**Comment by jeffcarp** [comment]
Hi @ninoCookies, thanks for using braintree-angular, hopefully I can help:

> the token is not needed for dropin situation

For Drop-In you do need to generate a `clientToken`.

> Also how would you receive the clientTokenPath constant if the server.js is running on different server for example?

`clientTokenPath` should work when passed a full URL https://github.com/jeffcarp/braintree-angular/blob/master/lib/braintree-factory.js#L23

tl;dr to your question: yes, you should be fine to push to production.

Longer answer: `braintree-angular` is not an officially supported Braintree module and has not undergone any security reviews. So my answer is "you should be fine," but with the caveat that I am not a security expert and therefore cannot guarantee the security of this module. I recommend reviewing all the code in this module (there isn't much) so you know what you're pushing to production.

**Comment by ninoCookies** [comment]
So how should i change the code to generate own tokens instead of dummy token that comes with example ? 
From what i checked /client-token path always generated unique token ? As you can see my questions are coming from perspective of not fully understanding everything so thank you for the patience and help so far :dancers:

**Comment by jeffcarp** [comment]
No worries at all, there's a lot going on in whole Braintree payment flow. Yes, the client-token should be unique each time you generate it.

I would start by deleting the `dummyClientToken` code: https://github.com/jeffcarp/braintree-angular/blob/master/examples/server.js#L25-L28

Have you filled in the gateway config with your sandbox credentials? e.g. https://github.com/jeffcarp/braintree-angular/blob/master/examples/server.js#L8

**Comment by ninoCookies** [comment]
Thank you for the suggestion. 

I filled the gateway config with sandbox credentials and ofcourse I see the payments coming in.. So everything is working for now. But now i have another problem... Since our payments are happening on a static page - we have no way of catching additional user information (Name, Lastname, Address) for delivery. I read braintree doc's about it and it seems pretty complicated so i'm thinking about adding my own database that will record this additional data ? Could you give me any pointers on how you resolved this - have you used **hosted fields**? I'm aware this topic is getting out of GitHub issues boundaries so just tell me and i'll close this.

**Comment by jeffcarp** [comment]
@ninoCookies you can continue using Drop-In for the same purpose if you integrate a DB. Hosted Fields would be a good candidate if you wanted the payment form to match the style of the other fields.

**Comment by ninoCookies** [comment]
Thank you for all the help - especially for question outside Braintree Angular plugin scope :D

We will indeed use hosted fields as you suggested.
